web/src/components/organisms/Settings/Integration/hooks.ts

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useMemo, useState } from "react";
+
+import {
+  Integration,
+  IntegrationMember,
+  Role,
+} from "@reearth-cms/components/molecules/Integration/types";
+import {
+  useGetMeQuery,
+  useAddIntegrationToWorkspaceMutation,
+  Role as GQLRole,
+  useUpdateIntegrationOfWorkspaceMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+
+export default (workspaceId?: string) => {
+  const [selectedConnectionModalIntegration, SetSelectedConnectionModalIntegration] =
+    useState<Integration>();
+  const [selectedIntegrationMember, SetSelectedIntegrationMember] = useState<IntegrationMember>();
+  const [integrationConnectModalShown, setIntegrationConnectModalShown] = useState(false);
+  const [integrationSettingsModalShown, setIntegrationSettingsModalShown] = useState(false);
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const { data, refetch } = useGetMeQuery();
+
+  const workspaces = data?.me?.workspaces;
+  const workspace = workspaces?.find(workspace => workspace.id === workspaceId);
+
+  const fromIntegration = (integration: any) => ({
+    id: integration.id,
+    name: integration.name,
+    description: integration.description,
+    logoUrl: integration.logoUrl,
+    developerId: integration.developerId,
+    developer: integration.developer,
+    iType: integration.iType,
+    config: {
+      token: integration.config?.token,
+      webhooks: integration.config?.webhooks,
+    },
+  });
+
+  const integrations = useMemo(() => {
+    return (data?.me?.integrations ?? [])
+      .map<Integration | undefined>(integration =>
+        integration ? fromIntegration(integration) : undefined,
+      )
+      .filter((integration): integration is Integration => !!integration);
+  }, [data?.me?.integrations]);
+
+  const workspaceIntegrationMembers = useMemo(() => {
+    return (workspace?.members ?? [])
+      .map<IntegrationMember | undefined>(member =>
```

Comment:
Again, please update all places that are using this style to the decided upon style (blah?.blah?.map())

Reply:
Done.

Reply:
https://github.com/reearth/reearth-cms/blob/58c8be6313ec45a489b63534dab11973f18ce352/web/src/components/organisms/Settings/Integration/hooks.ts#L49-L51